primo/data_parser/metric_data.py

```diff
@@ -0,0 +1,602 @@
+#################################################################################
+# PRIMO - The P&A Project Optimizer was produced under the Methane Emissions
+# Reduction Program (MERP) and National Energy Technology Laboratory's (NETL)
+# National Emissions Reduction Initiative (NEMRI).
+#
+# NOTICE. This Software was developed under funding from the U.S. Government
+# and the U.S. Government consequently retains certain rights. As such, the
+# U.S. Government has been granted for itself and others acting on its behalf
+# a paid-up, nonexclusive, irrevocable, worldwide license in the Software to
+# reproduce, distribute copies to the public, prepare derivative works, and
+# perform publicly and display publicly, and to permit others to do so.
+#################################################################################
+
+# Standard libs
+import logging
+import pandas as pd
+
+# User-defined libs
+from primo.utils.config_utils import UserPriorities
+from primo.data_parser.default_data import (
+    SUPP_IMPACT_METRICS,
+    SUPP_EFF_METRICS,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable = protected-access
+# pylint: disable = too-many-arguments
+# pylint: disable = too-many-instance-attributes
+class Metric:
+    """
+    Dataclass for creating metric objects
+    """
+
+    def __init__(
+        self,
+        name,
+        value,
+        min_value=0,
+        max_value=100,
+        full_name=None,
+    ) -> None:
+        """
+        Returns a `Metric` object
+
+        Parameters
+        ----------
+        name : str
+            Short name of the metric.
+            Must be a valid python variable name
+
+        value : float
```

Comment:
I suspect we might eventually have confusion regarding the terms value vs weight. 
 For example, consider a well which is not in a disadvantaged community and a scenario where the weight for the dac_impact metric is assigned as 25. Someone might easily expect the value attribute to return 0 but the weight attribute to return 25.

When we are thinking at the higher level of metrics, as I understand, we are only thinking of weights? Suggest therefore renaming:

- value -> weight
- min_value -> min_weight
- max_value -> max_weight and getting rid of the value getter and setter methods to avoid confusion

Reply:
Addressed by naming them `weight` and `effective_weight`